### PR TITLE
[BEAM-6439] Move Python Validates Runner Flink test to PreCommit.

### PR DESCRIPTION
**Pull request by robertwb**
The tests look fast and green now. 

https://builds.apache.org/view/A-D/view/Beam/view/PostCommit/job/beam_PostCommit_Python_VR_Flink/buildTimeTrend

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: .test-infra/jenkins/job_PreCommit_Python_ValidatesRunner_Flink.groovy [hunk 1]
```diff
@@ -16,25 +16,18 @@
...
-    }
-  }
-}
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+    scope: this,
+    nameBase: 'Python_ValidatesRunner_Flink',
+    gradleTask: ':beam-sdks-python:flinkValidatesRunner',
+    triggerPathPatterns: [
+      '^model/.*$',
+      '^runners/.*$',
```
Comment (mxm) [inline .test-infra/jenkins/job_PreCommit_Python_ValidatesRunner_Flink.groovy]: `runners/flink` and `runners/core-java`? Maybe easier simply to match all.

Comment (robertwb) [reply .test-infra/jenkins/job_PreCommit_Python_ValidatesRunner_Flink.groovy]: I think we'll want to at least catch changes in core-construction and java-fn-execution and as well. It's not pretty fast, so I think a blanket runners/... is fine.

**Comment by mxm** [comment]
Can be back this by a JIRA? Looks good otherwise.

**Comment by tweise** [comment]
The pre-commit coverage was already provided via https://builds.apache.org/job/beam_PreCommit_Portable_Python_Commit/

As we expect to expand test coverage, is it really necessary to jam all of this into pre-commit?

Also, this job was part of the post-commit display and now shows as disabled..
 
https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/

**Comment by kennknowles** [comment]
TBH I didn't look at code at all. If `PreCommitJobBuilder` takes that argument, then I presume yes.

**Comment by mxm** [comment]
I think having this in PreCommit is useful, so we can keep an eye on the status easily without having to run it for every PR. 

FYI Temporarily reverted: https://github.com/apache/beam/pull/7533

**Pull request merged**